src/VisualStudio/Core/Def/CodeCleanup/AbstractCodeCleanUpFixer.cs

```diff
@@ -193,11 +192,120 @@ private async Task<bool> FixAsync(
             var cancellationToken = context.OperationContext.UserCancellationToken;
             var progress = scope.GetCodeAnalysisProgress();
 
+            var initialSolution = workspace.CurrentSolution;
             var solution = await applyFixAsync(progress, cancellationToken).ConfigureAwait(true);
 
             await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
+            if (!IsSolutionValidForChanges(initialSolution, workspace.CurrentSolution, solution, out solution))
+            {
+                // Solution has changed. We can't apply this change.
+                return false;
+            }
+
             return workspace.TryApplyChanges(solution, progress);
+
+            bool IsSolutionValidForChanges(Solution initialSolution, Solution currentSolution, Solution changedSolution, out Solution updatedSolution)
+            {
+                updatedSolution = changedSolution;
+                if (initialSolution.WorkspaceVersion != currentSolution.WorkspaceVersion)
+                {
+                    // something has changed. Lets make sure we can still apply changes.
+
+                    var solutionChanges = initialSolution.GetChanges(currentSolution);
```

Comment:
To get changes from `currentSolution` we will be starting from `initialSolution`. See https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.solution.getchanges?view=roslyn-dotnet-4.9.0

```suggestion
                    var solutionChanges = currentSolution.GetChanges(initialSolution);
```